get-data.R

```diff
@@ -1,209 +1,175 @@
+# json_metadata_list[[1]]
+
+tidy_json_metadata <- function(x) {
+  x |>
+    select(any_of(
+      c(
+        "title",
+        "bcdc_resource_name",
+        "name",
+        "var_class",
+        "type",
+        "description",
+        "url"
+      )
+    ))
 }
 
-tidy_description <- function(x){
-  names(x)[names(x)=="field_description_and_notes"] <- "field_description"
-  }
-  
-## function to tidy the dfs
-tidy_metadata <- function(x){
-   
-  x %>% 
-    clean_names() %>%
-    rename_with(tidy_classification, .cols = matches("variable_classification")) %>% 
-    rename_with(tidy_description, .cols = matches("field_description_and_notes")) %>% 
-    select(
-      title,
-      bcdc_resource_name,
-      file_name,
-      field_name,
-      identifier_classification,
-      field_description,
-      url
-    )
+# map over JSON API metadata records
+df_metadata_json <- map_dfr(json_metadata_list,
+                            ~ tidy_json_metadata(.x))
+
+
+## Tidy DIP CSV API resources
+csv_metadata_list <- metadata_by_record |> 
+  keep(~("csv" %in% .$ext))
+
+tidy_csv_metadata <- function(x) {
+  x |>
+    clean_names() |> 
+    select(any_of(
+      c(
+        "title",
+        "bcdc_resource_name",
+        "name" = "field_name",
+        "var_class" = "variable_classification",
+        "var_class" = "identifier_classification",
+        "type" = "field_type",
+        "description" = "field_description",
+        "description" = "field_description_and_notes",
+        "url"
+      )
+    ))
 }
 
-# tidy_metadata(metadata_by_record[[2]])
-
-## map over metadata records and tidy the dfs
-df_metadata <- map_dfr(metadata_by_record, ~ tidy_metadata(.x))
+# map over CSV API metadata records
+df_metadata_csv <- map_dfr(csv_metadata_list,
+                            ~ tidy_csv_metadata(.x))
 
-
-## Collapse the code table rows to 1 per resource (since the attributes are all NA)
-code_files <- df_metadata %>%
-  filter(is.na(field_name)) %>%
-  group_by(bcdc_resource_name) %>% 
+## Collapse the csv "code table" rows to 1 per resource (since the attributes are all NA)
+code_files <- df_metadata_csv |> 
+  filter(is.na(name)) |> 
+  group_by(bcdc_resource_name) |> 
   slice(1L)
 
-code_names <- code_files %>% pull("bcdc_resource_name")
+code_names <- code_files |>  pull("bcdc_resource_name")
 
-df_metadata_reduced <-
-  df_metadata  %>%
-  filter(!bcdc_resource_name %in% code_names) %>%
+df_metadata_csv_reduced <-
+  df_metadata_csv  |> 
+  filter(!bcdc_resource_name %in% code_names) |> 
   bind_rows(code_files)
 
 
-## Final tidying step
-tidy_metadata <- df_metadata_reduced %>%
-  mutate(
-    identifier_classification = str_replace(identifier_classification, "�", "-"),
-    field_description = str_replace_all(field_description, "�", " ")
-  ) %>%
+## Join JSON+CSV tables and final tidying
+tidy_metadata <- df_metadata_csv_reduced |> 
+  bind_rows(df_metadata_json) |> 
   mutate(
-    data_provider = case_when(
```

Comment:
It's a shame this field is no longer useful! And a quick look at the catalogue doesn't show a different proxy that can be accessed.